### PR TITLE
Delete tag functionality 

### PR DESCRIPTION
**Pull request by NathanCenter**
#### Changes Made
1. Added file `DeleteTag.js` to `Modules` directory.
1. Modified file `TagController.cs` to include `public IActionResult Delete(int id)` and delete.
2. Modified file `ITagRepository.cs` to include `void Delete(int id);`
3.  Modified file `TagRepository.cs` to include `public Tag GetTagById(int id); and public void Delete(int id) `

#### Steps to Review
1. Checkout this branch locally.
    
    git fetch --all
    git checkout nc-deleteTag
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    > Instructions for how reviewer can test functionality, and detailed description of what the expected outcome is.
    > Example: When a user goes to tag management and clicks the delete button, then a new page should ask the user do you want to delete once you click the delete button it will go back to the tag list page with that tag gone.
6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
Add changes to code
1. TagDelete.js   delete the white space
2. TagView.js added a div that moves down the create a tag link a line.

### CODE REVIEW

File: Tabloid/client/src/components/TagDelete.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        })
+
+        
+          
+           
+      
+       
+          
+       
+
```
Comment (SchmidtFred) [inline Tabloid/client/src/components/TagDelete.js]: This is a lot of whitespace.

File: Tabloid/client/src/components/TagView.js [hunk 2]
```diff
@@ -22,9 +22,13 @@ export const Tags=()=>{
...
-            )} 
+           {tags?.map(t=>
+            <><h4 key={t.id}> {t.name}</h4><Link to={`/tag/delete/${t.id}`}> Delete</Link></>
+
+            
+            )
+            
+            }  
 
           <Link to="/tag/new">Create a tag</Link>
```
Comment (SchmidtFred) [inline Tabloid/client/src/components/TagView.js]: The last delete link is right next to create a tag. Is there a way to put create a tag underneath it?

File: Tabloid/client/src/components/TagDelete.js (hunk 1 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagDelete.js]: @SchmidtFred  delete the white space.

**Review by SchmidtFred — CHANGES_REQUESTED** [review]
Commented requested changes.

File: Tabloid/client/src/components/TagView.js (hunk 2 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagView.js]: @SchmidtFred added a div that moves down the create a tag link a line.

**Review by SchmidtFred — APPROVED** [review]

**Review by matt-rich-anderson — APPROVED** [review]
Looks good, watch those tags

**Pull request merged**